### ISSUE TITLE
Ask about the approach

### ISSUE THREAD

**Issue by KienPM** [wuch15/IJCAI2019-NAML#8]
Thank you for sharing this great repository!
Can you share me the reason why you consider article body as a long sequence instead of sentences. If  I want to encode each sentence then use sentences represent vector to encode article body, is it possible?

**Comment by wuch15** [comment]
Of course, you can use other architectures such as HAN to process the news body, and the performance is usually slightly better. But it usually requires a larger GPU memory/smaller batch size.

**Comment by KienPM** [comment]
Yeah, I've tried to encode each sentence then encode article body on 2080Ti GPU. I can only train with batch size = 1 and it took 25s/step, maybe something went wrong.
What does HAN stand for? Can you please share me a reference. Thank you very much!

**Comment by KienPM** [comment]
HAN stands for Hierarchical Attention Network, right?

**Comment by wuch15** [comment]
Yeah, HAN means Hierarchical Attention Network (Yang et al., 2016). You can replace the LSTM with CNN to boost the training speed.

**Comment by wuch15** [comment]
https://www.aclweb.org/anthology/N16-1174

**Comment by KienPM** [comment]
Yeup, thank you!
I've read some of your papers, they are awesome

**Comment by wuch15** [comment]
In addition, it is highly recommended that you can use a smaller sentence length or fewer sentences. Although I believe that using the full news body is beneficial, it takes a large amount of GPU memory and the improvement is usually marginal.

**Comment by KienPM** [comment]
Yeah, thanks for your recommendation
